### PR TITLE
Grahams merge

### PR DESCRIPTION
**Pull request by fullerga**
Features:
-not move outside the view of the camera
-enemies fade when they hit you and disappear
-the bounds on enemies match their shape and are much simpler now

### CODE REVIEW

**Comment by oakesja** [comment]
You can still move the player underneath the top blue background behind the healthbar. My suggestion is to just remove that entirely.

File: Assets/Scripts/Player.cs [hunk 1]
```diff
@@ -10,10 +10,17 @@ public class Player : MonoBehaviour
     private Plane plane;
     private HealthBar healthBar;
     private Color hitColor = new Color(1F, .15F, .15F, 1);
+    public float screenWidth;
```
Comment (oakesja) [inline Assets/Scripts/Player.cs]: These can be made private

File: Assets/Scripts/Player.cs [hunk 2]
```diff
@@ -10,10 +10,17 @@ public class Player : MonoBehaviour
...
+    public float screenHeight;
+
```
Comment (oakesja) [inline Assets/Scripts/Player.cs]: Extra whitespace

File: Assets/Scripts/Player.cs [hunk 3]
```diff
@@ -79,4 +84,24 @@ IEnumerator wait()
         yield return new WaitForSeconds(.3F);
         GetComponent<Renderer>().material.color = c;
     }
+
+    void LateUpdate()
+    {
+        if (transform.position.x > screenWidth)
+        {
```
Comment (oakesja) [inline Assets/Scripts/Player.cs]: These curly braces can be removed since the if statement is a single line.

File: Assets/Scripts/Player.cs [hunk 4]
```diff
@@ -10,10 +10,17 @@ public class Player : MonoBehaviour
...
+    public float screenHeight;
+
 
     void Start()
     {
         healthBar = GameObject.Find("health").GetComponent<HealthBar>();
+
+        Camera camera = GameObject.Find("MainCamera").GetComponent<Camera>();
+        screenWidth = camera.ScreenToWorldPoint(new Vector3(Screen.width, 0, 0)).x - .87F;
+        screenHeight = camera.ScreenToWorldPoint(new Vector3(0, Screen.height, 0)).y - .87F;
```
Comment (oakesja) [inline Assets/Scripts/Player.cs:22]: Why minus `.87F`? That seems pretty arbitrary?

File: Assets/Scripts/Player.cs [hunk 5]
```diff
@@ -46,9 +53,7 @@ void OnCollisionEnter2D(Collision2D col)
 
     private void HandleCollisionWithEnemy(Collision2D col)
     {
-        if (PowerUpManager.IsInvincible)
-            Destroy(col.gameObject);
```
Comment (oakesja) [inline Assets/Scripts/Player.cs:50]: Why did you remove this? This will break the invincible powerup.

File: Assets/Scripts/Enemies/EnemyMovement.cs [hunk 6]
```diff
@@ -28,6 +40,14 @@ private T FindGameObject<T>(string name)
         return obj.GetComponent<T>();
     }
 
+    void OnCollisionEnter2D(Collision2D col)
+    {
+        if (col.gameObject.name == "player")
+        {
```
Comment (oakesja) [inline Assets/Scripts/Enemies/EnemyMovement.cs]: Extra {'s that can be removed

File: Assets/Scripts/Enemies/EnemyMovement.cs [hunk 7]
```diff
@@ -18,6 +19,17 @@ void Start()
...
     {
         EnemyController.Update(gameObject, Time.deltaTime);
+
+        if (fading)
+        {
+            Color c = GetComponent<Renderer>().material.color;
+            GetComponent<Renderer>().material.color = new Color(c.r, c.g, c.b, c.a - 1.0F / 10);
+
+            if (c.a <= 0)
+            {
```
Comment (oakesja) [inline Assets/Scripts/Enemies/EnemyMovement.cs]: Extra {'s that can be removed

File: Assets/Scripts/Enemies/EnemyMovement.cs [hunk 8]
```diff
@@ -18,6 +19,17 @@ void Start()
     public virtual void Update()
     {
         EnemyController.Update(gameObject, Time.deltaTime);
+
+        if (fading)
+        {
+            Color c = GetComponent<Renderer>().material.color;
+            GetComponent<Renderer>().material.color = new Color(c.r, c.g, c.b, c.a - 1.0F / 10);
```
Comment (oakesja) [inline Assets/Scripts/Enemies/EnemyMovement.cs]: Consider moving the `1.0F / 10` to a const float at the top. This will allow us to easily tweak this without having to track it down in the file.

File: Assets/Scripts/Player.cs (hunk 5 above)
Comment (oakesja) [inline Assets/Scripts/Player.cs:50]: Nevermind I see why you removed it.

File: Assets/Scripts/Player.cs (hunk 4 above)
Comment (fullerga) [inline Assets/Scripts/Player.cs:22]: Because it works, I think it has to do with the white space on the png.
